backend/models.py

```
@@ -145,7 +145,7 @@ class Memecoin(Base):
     created_at = Column(DateTime, nullable=False, default=datetime.utcnow)
     
     # Relationships
-    positions = relationship('PortfolioPosition', backref='memecoin')
+    positions = relationship('PortfolioPosition', primaryjoin='Memecoin.symbol == PortfolioPosition.symbol', foreign_keys=[PortfolioPosition.symbol], viewonly=True, backref='memecoin_ref')
```

Flake8 (7.3.0)

[error] undefined name 'PortfolioPosition'

(F821)